Importers/Rhino/Speckle.Importers.Rhino/Program.cs

```diff
@@ -32,13 +37,34 @@ public static async Task Main(string[] args)
       TaskScheduler.UnobservedTaskException += (_, eventArgs) =>
         logger.LogCritical(eventArgs.Exception, "Unobserved Task Exception");
 
-      var importer = serviceProvider.GetRequiredService<ImporterInstance>();
+      var factory = serviceProvider.GetRequiredService<ImporterInstanceFactory>();
+
+      // Error handling flow below here looks a bit of a mess, but we're having to navigate threading issues with rhino inside
+      // https://discourse.mcneel.com/t/rhino-inside-fatal-app-crashes-when-disposing-headless-documents/208673/7
+      try
+      {
+        // This needs to be called on the main thread
+        importer = factory.Create(importerArgs);
```

Comment:
This is the main change. We're opening the document here on construction of the importer.
This way, we're doing it on the main thread.

Previously, this was done from a .NET worker thread inside the `Task.Run` block lower down.